frontend/src/metabase/public/containers/PublicDashboard/WithPublicDashboardEndpoints.tsx

```diff
@@ -0,0 +1,33 @@
+import type { ComponentType } from "react";
+import type { WithRouterProps } from "react-router";
+
+import {
+  setEmbedDashboardEndpoints,
+  setPublicDashboardEndpoints,
+} from "metabase/services";
+import type { DashboardId } from "metabase-types/api";
+
+export const WithPublicDashboardEndpoints = <T extends WithRouterProps>(
+  Component: ComponentType<T>,
+): ComponentType<T & { dashboardId: DashboardId }> => {
+  function DashboardEndpointsInner({
+    params: { uuid, token },
+    ...props
+  }: {
+    params: {
+      dashboardId?: DashboardId;
+      uuid?: string;
+      token?: string;
+    };
+  }) {
+    if (uuid) {
+      setPublicDashboardEndpoints();
+    } else if (token) {
+      setEmbedDashboardEndpoints();
+    }
+    const dashboardId = uuid || token;
+    return <Component {...(props as T)} dashboardId={dashboardId} />;
```

Comment:
Looks like `props.params.dashboardId` is never used? It's only the UUID and Token that is used.